armi/runLog.py

```diff
@@ -226,265 +195,346 @@ def setVerbosity(self, levelInput):
+                    print(rankId, file=sys.stderr)
+                    print(data, file=sys.stderr)
+            try:
+                os.remove(stderrName)
+            except OSError:
+                warning("Could not delete {0}".format(stderrName))
 
 
 # Here are all the module-level functions that should be used for most outputs.
-# They use the PrintLog object behind the scenes.
+# They use the Log object behind the scenes.
 def raw(msg):
     """
     Print raw text without any special functionality.
     """
-    LOG._emit(msg)  #  pylint: disable=protected-access
+    LOG.log("header", msg, single=False, label=msg)
 
 
 def extra(msg, single=False, label=None):
-    LOG.standardLogMsg("extra", msg, single=single, label=label)
+    LOG.log("extra", msg, single=single, label=label)
 
 
 def debug(msg, single=False, label=None):
-    LOG.standardLogMsg("debug", msg, single=single, label=label)
+    LOG.log("debug", msg, single=single, label=label)
 
 
 def info(msg, single=False, label=None):
-    LOG.standardLogMsg("info", msg, single=single, label=label)
+    LOG.log("info", msg, single=single, label=label)
 
 
 def important(msg, single=False, label=None):
-    LOG.standardLogMsg("important", msg, single=single, label=label)
+    LOG.log("important", msg, single=single, label=label)
 
 
 def warning(msg, single=False, label=None):
-    LOG.standardLogMsg("warning", msg, single=single, label=label)
+    LOG.log("warning", msg, single=single, label=label)
 
 
 def error(msg, single=False, label=None):
-    LOG.standardLogMsg("error", msg, single=single, label=label)
+    LOG.log("error", msg, single=single, label=label)
 
 
 def header(msg, single=False, label=None):
-    LOG.standardLogMsg("header", msg, single=single, label=label)
+    LOG.log("header", msg, single=single, label=label)
 
 
-def flush():
-    """Flush LOG's output in the err and output streams"""
-    LOG.flush()
+def warningReport():
+    LOG.warningReport()
 
 
-def prompt(statement, question, *options):
-    """ "Prompt the user for some information."""
-    from armi.localization import exceptions
+def setVerbosity(level):
+    LOG.setVerbosity(level)
+
+
+def getVerbosity():
+    return LOG.getVerbosity()
+
+
+# ---------------------------------------
 
-    if context.CURRENT_MODE == Mode.GUI:
-        # avoid hard dependency on wx
-        import wx  # pylint: disable=import-error
 
-        msg = statement + "\n\n\n" + question
-        if len(msg) < 300:
-            style = wx.CENTER
-            for opt in options:
-                style |= getattr(wx, opt)
-            dlg = wx.MessageDialog(None, msg, style=style)
+class DeduplicationFilter(logging.Filter):
+    """
+    Important logging filter
+
+    * allow users to turn off duplicate warnings
+        * This is done via some special string manipulation
+    * handles special indentation rules for our logs
+    """
+
+    singleMessageCounts = {}
+    singleWarningMessageCounts = {}
```

Comment:
Is there a reason why these are stored as class attributes? If so, it might warrant a comment explaining why

Reply:
Fixed